src/Yaapii.Xambly/Directive/NsDirective.cs

```diff
@@ -20,41 +20,138 @@
+        private readonly IText purpose;
+        private readonly bool inheritance;
+
+
+        /// <summary>
+        /// Namespace directive.
+        /// 
+        /// Sets namespace of all current nodes selected by the cursor.
+        /// All child nodes are moved to the namespace (default).
+        /// All attributes are moved to the namespace (default).
+        /// The namespace declaration will be done in the root node.
+        /// 
+        /// If the prefix is empty a default namespace will be created.
+        /// which is declared only in the current nodes.
+        /// Attributes cannot be added to a default namespace.
+        /// 
+        /// Hint:
+        /// After declaring a namespace the XPath will be affected.
+        /// </summary>
+        /// <param name="prefix">If empty a default namespace will be created</param>
+        /// <param name="ns">Namespace</param>
+        /// <param name="purpose">Set the namespace to: 'nodes', 'attributes', 'nodesAndAttributes'</param>
+        /// <param name="inheritance">Is applied to the children</param>
+        public NsDirective(string prefix, string ns, string purpose = DEFAULT_PURPOSE, bool inheritance = true) : this(
+            new Arg.AttributeArg(prefix),
+            new Arg.AttributeArg(ns),
+            purpose,
+            inheritance
+        )
+        { }
+
+        /// <summary>
+        /// Namespace directive.
+        /// 
+        /// Sets namespace of all current nodes selected by the cursor.
+        /// All child nodes are moved to the namespace (default).
+        /// All attributes are moved to the namespace (default).
+        /// The namespace declaration will be done in the root node.
+        /// 
+        /// If the prefix is empty a default namespace will be created.
+        /// which is declared only in the current nodes.
+        /// Attributes cannot be added to a default namespace.
+        /// 
+        /// Hint:
+        /// After declaring a namespace the XPath will be affected.
+        /// </summary>
+        /// <param name="prefix">If empty a default namespace will be created</param>
+        /// <param name="ns">Namespace</param>
+        /// <param name="purpose">Set the namespace to: 'nodes', 'attributes', 'nodesAndAttributes'</param>
+        /// <param name="inheritance">Is applied to the children</param>
+        public NsDirective(IArg prefix, IArg ns, string purpose = DEFAULT_PURPOSE, bool inheritance = true) : this(
+            new ScalarOf<string>(() => prefix.Raw()),
+            new ScalarOf<XNamespace>(() =>
+            {
+                XNamespace namesp = ns.Raw();
+                return namesp;
+            }),
+            purpose,
+            inheritance
+        )
         { }
 
-        public NsDirective(IArg nsp) : this(new Arg.AttributeArg(""), nsp)
+        private NsDirective(IScalar<string> prefix, IScalar<XNamespace> ns, string purpose, bool inheritance) : this(
```

Comment:
Replace purpose with two booleans so that devs do not need to learn strings.

Reply:
@Meerownymous 
Replace only in NsDirective internal but it is injectable as purpose when initializing the object?
OR
Replace at all?

My intention was to have a more readable code when using the NsDirective (or Directive) by injecting a human readable string instead of two Boolean.
What do you think?

Reply:
Yes when using the directive the code is more readable with a string that says `nodes` instead of two booleans
`new Directives().Ns("prefix", "nodes", true)` 
`new Directives().Ns("prefix", true, true, true)`
But i agree with @Meerownymous that for using the Ns directive with the string I have to learn the three strings `"node", "attributes", "nodesandattributes"`. With two booleans the coder does not need to learn the strings by reading the headerdocs but can code simply with the variable names.

Reply:
I would prefer this:
`new Directives().Ns("prefix", "nodes", "recursive")` (that changes the last bool to a string)
It is more readable when using the library as
`new Directives().Ns("prefix", true, false, true)`

If I understand it correctly, it is primarily about the readability of code. And the variant with strings is clearly more readable.
The fact that you have to know the strings when programming is not that important. With bool I also have to know what they mean. The effort is not very much different

Reply:
If you want to make it more readable when using bools, just write the variable prefix before it when using it. nodes: true etc.
Booleans cannot be misused because they are simply switches. But you can inject a wrong string. So booleans are the better design. 
If you want to enforce readable switches, you could introduce two Ns objects which you name accordingly. But I think that is kind of overthinking it.

Reply:
Okay, I will refactor it...